crates/uv/src/commands/pip_list.rs

```diff
@@ -70,6 +79,205 @@ pub(crate) fn pip_list(
         return Ok(ExitStatus::Success);
     }
 
+    if outdated {
+        let constraints: &[RequirementsSource] = Default::default();
+        let overrides: &[RequirementsSource] = Default::default();
+        let extras: ExtrasSpecification<'_> = ExtrasSpecification::default();
+        let connectivity: Connectivity = Connectivity::Online;
+
+        let resolution_mode: ResolutionMode = ResolutionMode::Highest;
+        let prerelease_mode: PreReleaseMode = PreReleaseMode::default();
+        let dependency_mode: DependencyMode = DependencyMode::default();
+        let upgrade: Upgrade = Upgrade::All;
+        let index_locations: IndexLocations = IndexLocations::default();
+        let keyring_provider: KeyringProvider = KeyringProvider::default();
+        let reinstall: Reinstall = Reinstall::All;
+        let setup_py: SetupPyStrategy = SetupPyStrategy::default();
+        let config_settings: ConfigSettings = ConfigSettings::default();
+        let no_build: NoBuild = NoBuild::None;
+        let no_binary: NoBinary = NoBinary::None;
+        let exclude_newer: Option<DateTime<Utc>> = None;
+        let break_system_packages: bool = false;
+        let native_tls: bool = false;
+
+        let _lock = venv.lock()?;
+
+        // Determine the set of installed packages.
+        let site_packages = SitePackages::from_executable(&venv)?;
+
+        let requirements = &site_packages
```

Comment:
We build up a list of requirements by taking all the currently installed packages and constructing `RequirementsSource::Package(name)` with no further version information, implying we want to resolve the latest version.